### ISSUE TITLE
Editor-Integrated Monitors

### ISSUE THREAD

**Issue by towerofnix** [ScratchAddons/ScratchAddons#1991]
**Describe the new addon**

Essentially a combo of my old requests LLK/scratch-gui#4117 and LLK/scratch-gui#4118, so I'll paste the relevant parts!

> When you drop a reporter block onto the stage, if it has a visibility toggle in the block palette (aka if you can show or hide it), a monitor for that reporter should be created wherever you dropped it.

> As a follow-up to (or part of) this feature, you should be able to drag a monitor back out of the stage and into the palette to delete it - the same intuitive gesture as removing a block. (This would be a handy feature even without the main idea in this issue, so that inexperienced users don't get confused and frustrated with monitors that they don't know how to get rid of!)

> What if dragging a monitor *from* the stage, *onto* the workspace (the blocks area), made a block show up where you dropped it? I think this would be a neat way to quickly take a variable or block you're actively using and keeping track of, y'know, 'cuz you're debugging uses of it or making a new feature with it, and use it in your code just like that. This isn't exactly an *obvious* use of monitors, but I think it would be really intuitive and handy.

These could be added as separatly toggle-able features if wanted, but they make the most sense when acting all together.

**Comment by Secret-chest** [comment]
This was in Scratch 0.5

**Comment by towerofnix** [comment]
@Secret-chest Yes it was! That's actually what inspired my original suggestions on the official LLK repositories, back in 2018. :)

**Comment by ksdavidc** [comment]
As a complement to this:

Right now if you right click on a reporter you have options (type, range, etc.).
What if if you right click on a non-reporter part of the stage, you get a drop down list of your variables (the same one used in blocks like `set [var v] to ()`) that you can then add or remove...

**Comment by towerofnix** [comment]
Maybe! It would be neat if it were a nested menu, like um...

* Show a Variable ->
  * my variable 1
  * my variable 2
  * etc
* Show a Property (or some-such)
  * x position
  * y position
  * volume
  * etc

But the menu system Scratch uses (\*cough\* [for the stage and "popup menu" GUI elements](https://github.com/scratchfoundation/scratch-gui/issues/1790) \*cough\*) doesn't support nested menus, I believe, so it would have to be quite the custom solution on Scratch Addons' part...
